atom/browser/native_window_mac.mm

```diff
@@ -1127,28 +1127,34 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
                                      const NativeWindow::ProgressState state) {
   NSDockTile* dock_tile = [NSApp dockTile];
 
+  // Sometimes macOS would install a default contentView for dock, we must
+  // verify whether NSProgressIndicator has been installed.
+  bool first_time = !dock_tile.contentView ||
+                    [[dock_tile.contentView subviews] count] == 0 ||
+                    ![[[dock_tile.contentView subviews] lastObject]
+                        isKindOfClass:[NSProgressIndicator class]];
+
   // For the first time API invoked, we need to create a ContentView in
   // DockTile.
-  if (dock_tile.contentView == nullptr) {
-    NSImageView* image_view = [[NSImageView alloc] init];
+  if (first_time) {
+    base::scoped_nsobject<NSImageView> image_view([[NSImageView alloc] init]);
```

Comment:
I don't think there's much point in using `scoped_nsobject` if you're going to immediately autorelease it. Just:
```suggestion
    NSImageView* image_view = [[[NSImageView alloc] init] autorelease];
```